homeassistant/components/media_player/cast.py

```diff
@@ -224,8 +225,12 @@ def _async_create_cast_device(hass: HomeAssistantType,
         info = ChromecastInfo(host=discovery_info['host'],
                               port=discovery_info['port'])
     elif CONF_HOST in config:
+        if CONF_PORT in config:
+            ccport = config[CONF_PORT]
+        else:
+            ccport = DEFAULT_PORT
```

Comment:
Add the default in the voluptuous schema, like `CONF_IGNORE_CEC`

Reply:
So, after a lot of catching up on the dev docs and guidelines, I agree this was indeed a too quick & dirty fix (after all, I just customized it for my personal needs and thought I'd just PR it as well).

Adding the default to the schema does not work. As far as I have understood, it is because of the following:
https://github.com/home-assistant/home-assistant/blob/f929c38e980d728bc74055043310bdc4c24211a1/homeassistant/components/media_player/cast.py#L201-L209

This is set by the cast platform, which did not pass through the schema:
https://github.com/home-assistant/home-assistant/blob/2f2bcf0058e9ce2e56cc06e8cf6786b44fa6c866/homeassistant/components/cast/__init__.py#L10-L14

Because of this, config[CONF_PORT] doesn't exist at all, instead of being supplied with its default value. I think this is a side-effect from when cast was moved to its own platform instead of being solely a media_player component?

An easy, but, granted, dirty fix would be to fill the CONF_PORT key in the wrapping function async_setup_entry(), or in the platform setup. I see lot of components are doing this, but it would be basically the same as my initial PR, so, not sure how to proceed here.